libs/core/src/main/java/org/opensearch/core/common/io/stream/StreamInput.java

```diff
@@ -1305,6 +1305,21 @@ public <E extends Enum<E>> EnumSet<E> readEnumSet(Class<E> enumClass) throws IOE
         return res;
     }
 
+    /**
+     * Reads an optional enum with type E that was serialized based on the value of its ordinal
```

Comment:
This sentence implies a single enum and doesn't match the reading of a set with multiple members.

Reply:
sure, edited to add clarity